### ISSUE TITLE
Dart 2 support

### ISSUE THREAD

**Issue by aabounegm** [Pajn/Neo4jDart#18]
This driver is very useful, but unfortunately, it is not compatible with Dart 2. Are there any plans to support Dart 2?

**Comment by zjwarnes** [comment]
is it worth while to update this driver to support dart 2 ourselves? The original creator doesn't seem very active, I don't expect to see a response

**Comment by aabounegm** [comment]
@zjwarnes 
I think so. I already took a shot at it and it was relatively simple except that most of its dependencies are also not compatible with Dart 2. Some of them are not crucial (such as testing libraries), but others are important, including the other library written by the same author (https://github.com/Pajn/Warehouse). It required more work, and I didn't spend more than a day on it because my need for it got postponed a little.
Seeing as the author is active on other repos, it seems they might have lost interest in this particular project, but I hope they would approve a pull request if it is working.

**Comment by Pajn** [comment]
Hi,

I don't use Dart much at all anymore. If someone takes the time to update this to Dart 2 though I can transfer the ownership of the pub packages to that person (or if that's not possible, accept a PR and publish myself)

Warehouse uses reflection/mirrors quite a lot. I don't know if that's supported in Dart 2 or if it's just Flutter that does not. Without mirrors support I don't think it would be worthwhile to update warehouse instead of creating or finding a replacement to it.

**Comment by aabounegm** [comment]
@Pajn Thank you for the response and the willingness to transfer ownership. I am willing to give it a shot, but unfortunately, I started the new semester in university and don't have that much time on my hand. I will try at it whenever I have free time.
Dart 2 does support reflection and mirrors: https://api.dart.dev/stable/2.7.1/dart-mirrors/dart-mirrors-library.html, although its status is **unstable**. With Warehouse, there are a few problems having to do with static typing and generic type parameters, but only in 2 files. I hope to be able to resolve them after I take a deep look into the package and understand what everything does.
@zjwarnes I am willing to collaborate on this and already created a fork where I played around a little (didn't push anything, though).

**Comment by zjwarnes** [comment]
@aabounegm I'm in the same position as you, I'm currently starting the new semester. However, I have a project that should align with this work so I'm going to help as much as I can.

**Comment by DeGerlash** [comment]
Has any progress been made on a port to Dart 2 mean the while? I'd be happy to help if someone has already created a public fork somewhere, I've ran into the exact same issue as you @aabounegm, my app requires a graph database backend but there's no Dart 2 support for it.

**Comment by aabounegm** [comment]
@DeGerlash not really. I forked the repo back when I attempted to convert it, but there were too many dependencies (some 1st party, some not) that also needed updating, and the task at my job changed such that we no longer needed Neo4J so I stopped the attempts. I'd be happy to give it another shot, but I don't really have enough time on my hands these days because of the studies.

**Comment by DeGerlash** [comment]
For anyone wandering around this thread in the future: neo4j has a generic HTTP API that can be used from any language. For simple projects or POC's, this might be enough. It definately is for me.
https://neo4j.com/developer/language-guides/#http-api
